### DISCUSSION TITLE
Open global search on the current buffer directory

### DISCUSSION THREAD

**Discussion by rochacbruno** [helix-editor/helix#8193] [Ideas] [answered]
Hi,

There is a way to open file_picker_in_current_buffer_directory which I mapped to <space>fb and is very helpful because it loads faster than <space>f/F if I know I want a file from the very same dir.

This is specially powerful in the situation like:

```console
cd
~$/ helix ~/.config/helix/config.toml
```
In the above example the CWD is `/home/user` but the file opened is on a subdirectory

with that file opened if I do <space>/ to open global_search an type a very common token such as `a` and hit enter.

Helix hangs for almost 10 minutes (in my system) because it tries to fuzzy find on my entire `/home/user` directory

## What I wanted?

Ability to map `global_search_in_current_buffer_directory` so the fuzzy finder would index and search only child nodes from that directory.

## Workaround

I  tried to map a chain of commands like

```
<space>fs = [":cd CURRENT_BUFFER_DIR", "global_search", "cd RESTORE_LATEST_CWD"]
```

However I don´t  know how to fill CURRENT_BUFFER_DIR and RESTORE_LATEST_CWD

I appreciate any suggestion

**Comment by David-Else** [comment]
```toml
[keys.normal.space]
"." = "file_picker_in_current_buffer_directory"
```

**Reply by kirawi** [reply]
The question is about `space-/` in only the current buffer directory.

**Reply by rochacbruno** [reply]
Thanks @David-Else I am already mapping this command, but I am looking for similar command for the global search

**Reply by pascalkuthe** [reply]
I mean that would just be search `/` a picker doesn't really make too much sense since the global search picker just matches on filenames (but in this case all filenames are the same)

**Reply by rochacbruno** [reply]
@pascalkuthe the global search `<space>/` searches for a term in all the files in the working directory, what I am looking for is to scope the search to the  current buffer dir only

**Reply by the-mikedavis** [reply, marked as answer]
@pascalkuthe and I chatted about this and we don't want to end up with a bunch of variants of each picker command per-scope (i.e. `global_search_in_workspace`, `global_search_in_current_working_directory`, `global_search_in_current_buffer_directory`). Rather we think variables and typable commands are probably the best way to handle this, so we would have `:global-search` in the workspace, `:global-search %{cwd}` for current working directory and `:global-search %{dirname}` to handle these cases with variable expansions, which should be keymappable. We will need variable expansion and, for global_search specifically, the refactor for #196 to make this possible.

**Comment by JakubKoralewski** [comment]
Any progress on this? Space + . for 'Search at current buffer's directory', or global search with `search tearm %p %{dirname}` would be nice
